### DISCUSSION TITLE
Test failures - maya_usd-0.6.0 + usd-21.02 + maya-2020 (Linux)

### DISCUSSION THREAD

**Discussion by pmolodo** [Autodesk/maya-usd#1019] [Q&A] [answered]
Hi - I'm getting some test failures when building with this combination:

Linux (Centos 7.7)
maya_usd-0.6.0 (tagged)
usd-21.02 ([latest dev e9172af0](https://github.com/PixarAnimationStudios/USD/commit/e9172af0ccea9896990f5ac093c3d6063ac7cab5))
maya-2020.3

The tests that fail are:

```
         18 - testProxyShapeLiveSurface (Failed)
         20 - testProxyShapeSelectionPerformance (Failed)
         97 - testUsdExportCustomConverter-Maya (Failed)
         98 - testUsdExportCustomConverter-USD (Failed)
         99 - testUsdImportCustomConverter-Maya (Failed)
        100 - testUsdImportCustomConverter-USD (Failed)
```

Can anybody else confirm that they do/don't get these failures?

**Comment by mattyjams** [comment]
Hey @pmolodo!

I just gave that configuration a try and I am not seeing any test failures. I'm running on RHEL 7.9, but other than that, the core USD, maya-usd, and Maya versions were all as you specified.

Any details in the log as to where those tests are failing?

**Reply by pmolodo** [reply]
Hmm, ok, thanks for checking.  I'll dig into it a bit more to figure out what's messed up on my end...

**Comment by pmolodo** [comment]
TIL: trailing whitespace is important in maya `.mod` files:
A file with this line will work:

`"scripts: \n"`

whereas this won't, and will cause the entire section of the .mod file to be silently ignored:

`"scripts:\n"`

(At least as of maya-2019. Haven't checked 2020 yet...)

**Comment by seando-adsk** [comment]
@pmolodo I checked all the .mod files in the repo and they all have a trailing space after the colon (**:**) so they should be good. But even so when running the tests we don't use the .mod files, instead the environment is setup by the cmake test calls. Is there something to fix in the repo, or was is something specific to how you are running the tests?

Sean

**Reply by pmolodo** [reply]
Yeah, sorry, should have been more specific - the .mod thing was a problem with my repo.  (At some point, I made an edit, and my auto-strip-trailing-whitespace option was on.)
Now sure why my tests were using the .mod files -it may be because of custom alterations to the testing environment on our end.  I'll investigate more when I get a chance...

**Comment by pmolodo** [comment, marked as answer]
So, just to close the loop on this discussion - my failures were caused by a combination of two things:

- usage of PXR_OVERRIDE_PLUGINPATH_NAME
  - this caused the testUsdExportCustomConverter* tests to fail
  - now fixed by https://github.com/Autodesk/maya-usd/pull/1108 ()
- the fact that the .mod was on my `MAYA_MODULE_PATH`
  -  ...which made the `VP2_RENDER_DELEGATE_PROXY` env var get set
  - this caused the testProxyShape* tests to fail
  - now also fixed in 0.7.0, due to the fact that we've switched to opt-out, via `MAYAUSD_DISABLE_VP2_RENDER_DELEGATE`, and the tests now set this env-var.

Re: the second issue, related to the `.mod` file causing tests to fail... this does make me think that we should have some way of testing the .mod.  Ideally, it would be used during ALL the tests, instead of manually setting env vars for them... but I'm not sure how difficult that would be - ie, I think the tests currently run out of the build dir, while the `.mod` is set up to use the install dir.  Perhaps we could make the tests require an install, or perform a "temp" install, within the build dir, somehow?
